### PR TITLE
Zoma/cross train1

### LINKED ISSUE #2893: Ensure adding QnA properly adds cross train recognizer
In order for a dialog to be loaded as part of tests all dependent dialogs must be present which is a problem when we rely on build for adding recognizers.  This was fixed for standard schema by generating the multi-lingual recognizer as a template and removing that responsibility from luconfig.json.  I suspect that we need to do the same thing for the cross-train recognizer when QnA is added.  We also need some tests that include professional schema.

### PR DESCRIPTION
**Pull request by coldplaying42**
Fixes #2893 

## Proposed Changes

  - 
  -
  -

### CODE REVIEW

File: experimental/generation/generator/packages/library/templates/professional/config.json.lg [hunk 1]
```diff
@@ -11,7 +11,7 @@ ${join(foreach(locales, locale, localeTemplate(locale)), ','+EOL())}
 ```
 
 # localeTemplate(locale)
--```    "./${knowledgeDir()}${locale}/${prefix}.${locale}.lu": {
+-```    "${prefix}.${locale}": {
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/professional/config.json.lg:14]: >${prefix} [](start = 9, length = 9)

Why did you change this?  For composer this is where the results need to be.

File: experimental/generation/generator/packages/library/templates/standard/build.cmd.lg [hunk 2]
```diff
@@ -45,8 +45,8 @@ ${join(foreach(locales, locale, copyForLocale(locale)) , EOL())}
 
 # copyForLocale(locale)
 -```
-call copy .\cross-trained\\${prefix}.${locale}.lu .\\${locale}\ /y
-call copy .\cross-trained\\${prefix}.${locale}.qna .\\${locale}\ /y
+call copy .\cross-trained\\${prefix}.${locale}.lu .\language-understanding\\${locale}\ /y
+call copy .\cross-trained\\${prefix}.${locale}.qna .\knowledge-base\\${locale}\ /y
 ```
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/build.cmd.lg:50]: Wouldn't this be better in the template filename rather than done here?

**Review by chrimc62 — CHANGES_REQUESTED** [review]
:clock1:

File: experimental/generation/generator/packages/library/templates/standard/build.cmd.lg (hunk 2 above)
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/build.cmd.lg:50]: This would be better in the template filename rather than relying on this script which will not be run by composer.

File: experimental/generation/generator/packages/library/templates/professional/config.json.lg [hunk 3]
```diff
@@ -11,7 +11,7 @@ ${join(foreach(locales, locale, localeTemplate(locale)), ','+EOL())}
...
             "rootDialog": true,
 		    "triggers": {"":""}
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/professional/config.json.lg:16]: >	    "triggers": {"":""} [](start = 1, length = 24)

"out": ${knowledgeDir}

File: experimental/generation/generator/packages/library/templates/professional/config.json.lg (hunk 1 above)
Comment (coldplaying42) [reply experimental/generation/generator/packages/library/templates/professional/config.json.lg:14]: I asked Fei, he said in Composer, the config file for cross train will be generated automatically.  Could we keep the change?

**Review by chrimc62 — APPROVED** [review]
:shipit:
